src/gui/qgsdbrelationshipwidget.cpp

```diff
@@ -0,0 +1,245 @@
+        } )
+  {
+    if ( cardinalities.contains( cardinality ) )
+      mCardinalityCombo->addItem( QgsRelation::cardinalityToDisplayString( cardinality ), QVariant::fromValue( cardinality ) );
+  }
+
+  const QList<Qgis::RelationshipStrength> strengths = mConnection->supportedRelationshipStrengths();
+  for ( Qgis::RelationshipStrength strength :
+        {
+          Qgis::RelationshipStrength::Association,
+          Qgis::RelationshipStrength::Composition
+        } )
+  {
+    if ( strengths.contains( strength ) )
+      mStrengthCombo->addItem( QgsRelation::strengthToDisplayString( strength ), QVariant::fromValue( strength ) );
+  }
+
+  const Qgis::RelationshipCapabilities capabilities = mConnection->supportedRelationshipCapabilities();
+  if ( !( capabilities & Qgis::RelationshipCapability::ForwardPathLabel ) )
+  {
+    mForwardLabelLineEdit->hide();
+    mForwardLabel->hide();
+  }
+
+  if ( !( capabilities & Qgis::RelationshipCapability::BackwardPathLabel ) )
+  {
+    mBackwardLabelLineEdit->hide();
+    mReverseLabel->hide();
+  }
+
+  const QStringList relatedTableTypes = mConnection->relatedTableTypes();
+  mRelatedTableTypeCombo->addItems( relatedTableTypes );
+
+  mProxyModel = new QSortFilterProxyModel( mTableModel );
+  mProxyModel->setSourceModel( mTableModel );
+  mProxyModel->setDynamicSortFilter( true );
+  mProxyModel->setSortCaseSensitivity( Qt::CaseInsensitive );
+  mProxyModel->setSortRole( Qt::DisplayRole );
+  mProxyModel->sort( 0 );
+
+  mLeftTableCombo->setModel( mProxyModel );
+  mRightTableCombo->setModel( mProxyModel );
+
+  connect( mNameEdit, &QLineEdit::textChanged, this, [ = ]
+  {
+    emit validityChanged( isValid() );
+  } );
+  connect( mLeftTableCombo, qOverload< int >( &QComboBox::currentIndexChanged ), this, [ = ]( int )
+  {
+    mLeftFieldsCombo->setFields( mConnection->fields( QString(), mLeftTableCombo->currentText() ) );
+    emit validityChanged( isValid() );
+  } );
+  connect( mRightTableCombo, qOverload< int >( &QComboBox::currentIndexChanged ), this, [ = ]( int )
+  {
+    mRightFieldsCombo->setFields( mConnection->fields( QString(), mRightTableCombo->currentText() ) );
+    emit validityChanged( isValid() );
+  } );
+  connect( mCardinalityCombo, qOverload< int >( &QComboBox::currentIndexChanged ), this, [ = ]( int )
+  {
+    emit validityChanged( isValid() );
+  } );
+  connect( mLeftFieldsCombo, qOverload< int >( &QComboBox::currentIndexChanged ), this, [ = ]( int )
+  {
+    emit validityChanged( isValid() );
+  } );
+  connect( mRightFieldsCombo, qOverload< int >( &QComboBox::currentIndexChanged ), this, [ = ]( int )
+  {
+    emit validityChanged( isValid() );
+  } );
+  connect( mStrengthCombo, qOverload< int >( &QComboBox::currentIndexChanged ), this, [ = ]( int )
+  {
+    emit validityChanged( isValid() );
+  } );
+  connect( mRelatedTableTypeCombo, qOverload< int >( &QComboBox::currentIndexChanged ), this, [ = ]( int )
+  {
+    emit validityChanged( isValid() );
+  } );
```

Comment:
could perhaps be modified as a loop over { mCardinaityCombo, mLeftFieldsCombo, ... , mRelatedTableTypeCombo } to decrease a bit code verbosity

Reply:
done